packages/components/src/Sales/Auction/Form.tsx

```diff
@@ -165,7 +163,21 @@ const SalesAuctionForm: VFC<Props> = ({
               <NumberInputField
                 id="price"
                 placeholder={t('sales.auction.form.price.placeholder')}
-                {...register('price')}
+                {...register('price', {
+                  validate: (value) => {
+                    const splitValue = value.split('.')
+
+                    if (parseFloat(value) > 0) {
+                      if (splitValue[1].length > currency.decimals) {
+                        return t('sales.auction.form.validation.decimals', {
+                          nbDecimals: currency.decimals,
+                        })
+                      }
+                    } else {
+                      return t('sales.auction.form.validation.positive')
+                    }
```

Comment:
Same comment about doing the condition in the other way:
```suggestion
                    if (parseFloat(value) <= 0)
                      return t('sales.auction.form.validation.positive')
                    if (splitValue[1].length > currency.decimals)
                      return t('sales.auction.form.validation.decimals', {
                        nbDecimals: currency.decimals,
                      })
```
Also `splitValue[1]` may be undefined if the number doesn't contain a `.`:
```
if (splitValue[1] && splitValue[1].length > currency.decimals)
```